waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/ThriftMetastoreClient.java

```diff
@@ -217,15 +218,23 @@ public void close() {
     }
     // Transport would have got closed via client.shutdown(), so we don't need this, but
     // just in case, we make this call.
-    if (isOpen()) {
+    if (transport != null && transport.isOpen()) {
       transport.close();
       transport = null;
     }
     LOG.info("Closed a connection to metastore, current connections: " + CONN_COUNT.decrementAndGet());
   }
 
   public boolean isOpen() {
-    return transport != null && transport.isOpen();
+    if (transport != null && transport.isOpen() && client != null) {
+      try {
+        client.getStatus();
```

Comment:
more or less just picked some call that can be made, doesn't really matter this one seems fairly lightweight